peppy/project.py

```diff
@@ -441,20 +441,67 @@ def templates_folder(self):
         return os.path.join(os.path.dirname(__file__), "submit_templates")
 
 
-    @staticmethod
-    def infer_name(path_config_file):
+    def infer_name(self):
         """
-        Infer project name based on location of configuration file.
+        Infer project name from config file path.
+        
+        First assume the name is the folder in which the config file resides,
+        unless that folder is named "metadata", in which case the project name
+        is the parent of that folder.
+        
+        :param str path_config_file: path to the project's config file.
+        :return str: inferred name for project.
+        """
+        import os
+
+        if hasattr(self, "name"):
+            return(self.name)
+        
+        config_folder = os.path.dirname(self.config_file)
+        project_name = os.path.basename(config_folder)
+        
+        if project_name == "metadata":
+            project_name = os.path.basename(os.path.dirname(config_folder))
+
+        self.name = project_name
+
+        return self.name
+
+    def get_subsample(self, sample_name, subsample_name):
```

Comment:
Would prefer to keep with the (mostly?) consistent double-spacing between member function definitions.

Reply:
yeah, my bad.